mlir/test/Dialect/Arith/canonicalize.mlir

```diff
@@ -2950,6 +2950,14 @@ func.func @unsignedExtendConstantResource() -> tensor<i16> {
   return %ext : tensor<i16>
 }
 
+// Just checks that this doesn't crash.
+// CHECK-LABEL: @signedExtendSplatAsDynamicShape
+func.func @signedExtendSplatAsDynamicShape() -> tensor<?xi64> {
+  %splat = arith.constant dense<5> : tensor<2xi16>
+  %extsplat = arith.extsi %splat : tensor<2xi16> to tensor<?xi64>
```

Comment:
I'm surprised this is even allowed at the level of `artith`. Shouldn't we change the verifier to reject this extend + 'shape cast' pattern instead?

Reply:
Makes sense. I'll rewrite it so that it's rejected during verification.

Reply:
That might be a different can of worms. This is an obvious things to fix for now. Maybe we should land this first.